enterprise/api/v1/views.py

```diff
@@ -456,6 +456,27 @@ class EnterpriseCustomerReportingConfigurationViewSet(EnterpriseReadOnlyModelVie
     ordering_fields = FIELDS
 
 
+class CatalogQueryView(APIView):
+    """
+    View for enterprise catalog query.
+
+    This will be called from django admin tool to populate `content_filter` field of `EnterpriseCustomerCatalog` model.
+    """
+    authentication_classes = [SessionAuthentication]
+    permission_classes = [permissions.IsAuthenticated, permissions.IsAdminUser]
+    http_method_names = ['get']
+
+    def get(self, request, catalog_query_id):
+        """
+        API endpoint for fetching a enterprise catalog query.
+        """
+        try:
+            catalog_query = models.EnterpriseCatalogQuery.objects.get(pk=catalog_query_id)
+        except models.EnterpriseCatalogQuery.DoesNotExist:
+            return Response({"detail": "Could not find enterprise catalog query."}, status=HTTP_404_NOT_FOUND)
```

Comment:
Do we need any logging here?

Reply:
I don't think this is critical place to monitor plus we will hardly get this error.